### PR TITLE
feat: Bump OTel image version to 0.115.0

### PR DESCRIPTION
**Pull request by hisarbalik**
## Description

Changes proposed in this pull request (what was done and why):

- Bump OTel image version to 0.115.0
- Bump telemetrygen image version to 0.115.0
- Add benchmark test result for OTel image version 0.115.0

Changes refer to particular issues, PRs or documents:

- https://github.com/kyma-project/opentelemetry-collector-components/pull/177

## Traceability
- [x] The PR is linked to a GitHub issue.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [x] If the change is user-facing, the documentation has been adjusted.
- [ ] If a CRD is changed, the corresponding Busola ConfigMap has been adjusted.
- [x] The feature is unit-tested.
- [x] The feature is e2e-tested.

### CODE REVIEW

File: test/testkit/mocks/backend/deployment.go [hunk 1]
```diff
@@ -11,7 +11,7 @@ import (
 )
 
 const (
-	otelCollectorImage = "europe-docker.pkg.dev/kyma-project/prod/kyma-otel-collector:0.114.0-main"
+	otelCollectorImage = "europe-docker.pkg.dev/kyma-project/prod/kyma-otel-collector:0.115.0-main"
```
Comment (a-thaler) [inline test/testkit/mocks/backend/deployment.go:14]: Can we maybe take the value from the main.go constants as well to reduce the amount of files to touch?

File: test/testkit/mocks/telemetrygen/telemetrygen.go [hunk 2]
```diff
@@ -124,7 +124,7 @@ func PodSpec(signalType SignalType, opts ...Option) corev1.PodSpec {
 		Containers: []corev1.Container{
 			{
 				Name:  "telemetrygen",
-				Image: "ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen:v0.114.0",
+				Image: "ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen:v0.115.0",
```
Comment (a-thaler) [inline test/testkit/mocks/telemetrygen/telemetrygen.go:127]: Can we maybe take the value from the main.go constants as well to reduce the amount of files to touch?

**Comment by a-thaler** [comment]
You missed the load test, there we also use an image and telemetrygen.. how can we simplify that?

**Review by NHingerl — APPROVED** [review]

File: test/testkit/mocks/telemetrygen/telemetrygen.go (hunk 2 above)
Comment (hisarbalik) [reply test/testkit/mocks/telemetrygen/telemetrygen.go:127]: I will create a proposal for using a single configuration/constant package for such values/cases

**Review by a-thaler — APPROVED** [review]

**Pull request merged**